provider/server/exec_test.go

```diff
@@ -30,9 +38,12 @@ var _ = Describe("Exec", func() {
 		Expect(createResp).NotTo(BeNil())
 
 		DeferCleanup(func(ctx SpecContext) {
-			_, err := machineClient.DeleteMachine(ctx, &iri.DeleteMachineRequest{MachineId: createResp.Machine.Metadata.Id})
-			Expect(err).ShouldNot(HaveOccurred())
 			Eventually(func() bool {
+				_, err := machineClient.DeleteMachine(ctx, &iri.DeleteMachineRequest{MachineId: createResp.Machine.Metadata.Id})
```

Comment:
pls call DeleteMachine just one time before Eventually

Reply:
We can not take it out of Eventually, we were facing this machine object getting overridden problem sometimes as mentioned in #164 . In all other integration test also its been handled similar way. May be after this fix we can tryout.